packages/infolists/src/Components/Concerns/HasIconColor.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Filament\Infolists\Components\Concerns;
+
+use Closure;
+
+trait HasIconColor
+{
+    /**
+     * @var string | array{50: string, 100: string, 200: string, 300: string, 400: string, 500: string, 600: string, 700: string, 800: string, 900: string, 950: string} | Closure | null
+     */
+    protected string | array | Closure | null $iconColor = null;
+
+    /**
+     * @param  string | array{50: string, 100: string, 200: string, 300: string, 400: string, 500: string, 600: string, 700: string, 800: string, 900: string, 950: string} | Closure | null  $color
+     */
+    public function iconColor(string | array | Closure | null $color): static
+    {
+        $this->iconColor = $color;
+
+        return $this;
+    }
+
+    /**
+     * @return string | array{50: string, 100: string, 200: string, 300: string, 400: string, 500: string, 600: string, 700: string, 800: string, 900: string, 950: string} | null
+     */
+    public function getIconColor(mixed $state): string | array | null
+    {
+        $color = $this->evaluate($this->iconColor, [
+            'state' => $state,
+        ]);
+
+        if ($color === false) {
+            return null;
+        }
+
+        if (filled($color)) {
+            return $color;
+        }
```

Comment:
Do we really need these checks, @aymanalareqi @danharrin?

Reply:
No